plugins/osdn/osdn.go

```diff
@@ -207,6 +207,11 @@ func (oi *OsdnRegistryInterface) WriteNetworkConfig(network string, subnetLength
 	if err == nil {
 		if cn.Network == network && cn.HostSubnetLength == int(subnetLength) && cn.ServiceNetwork == serviceNetwork {
 			return nil
+		} else if cn.Network == network && cn.HostSubnetLength == int(subnetLength) && cn.ServiceNetwork == "" {
+			// Upgrade from 3.0.0
+			cn.ServiceNetwork = serviceNetwork
+			_, err = oi.oClient.ClusterNetwork().Update(cn)
```

Comment:
You're supposed to be able to have more than one master running at once? I don't think the openshift-sdn code expects/supports that at all. Eg, in WriteNetworkConfig(), if the record doesn't already exist both masters would try to Create() it at the same time a few lines below, and one of them would fail, right? So the function is already broken in that case (along with basically every other place where the sdn master writes a value to etcd...)